### PR TITLE
fix some issues around broadcast profiles

### PR DESCRIPTION
**Pull request by iamcodemaker**
This fixes the following issues:
- crash when deleting profiles (introduced in 10ec4cb).
- proper handling of profiles with dot in the name
- proper handling of profiles with forbidden characters in the name

Previously the code assumed the profile name was stored in the xml file but this was not the case. This has been fixed. Additionally, the code would derive the file name from the profile name which doesn't work if the two don't match (like in the previously mentioned case with a `.` in the name). Now the filename is tracked separately.

There are still some edge cases that may cause unexpected behavior if a user previously had profiles with dots or special characters in the name. Specifically, the profile name in the GUI won't match the file name and it will be possible to have several profiles with the same name  (the last one loaded will be used). This should only happen if a user previously saved a profile with a dot (`.`) in the name. Going forward, any new or edited profiles should behave as expected.

### CODE REVIEW

File: src/preferences/broadcastsettings.cpp [hunk 1]
```diff
@@ -199,12 +191,17 @@ void BroadcastSettings::applyModel(BroadcastSettingsModel* pModel) {
     // TODO(Palakis): lock both lists against modifications while syncing
 
     // Step 1: find profiles to delete from the settings
-    for(const auto& actualProfile : m_profiles) {
-        QString profileName = actualProfile->getProfileName();
+    for(auto actualProfile = m_profiles.begin(); actualProfile != m_profiles.end();) {
```
Comment (iamcodemaker) [inline src/preferences/broadcastsettings.cpp]: we could also use `m_profiles.values()` here, but using `erase()` is slightly more efficient.

Comment (uklotzde) [reply src/preferences/broadcastsettings.cpp]: Even as a non-native speaker I suppose the term "actual" is wrong here. We should replace it. Since it is an iterator you may use "iterProfile" or whatever. I don't like type prefixes, but in this case it might be helpful to distinguish the iterator from the actual type, i.e. "removedProfile". Maybe "missingProfile" or "orphanedProfile" would be a more appropriate naming for this variable, because this is what we actually know about it before we are going to delete it.

Please add a space after the keyword _for_ to distinguish it from a function call.

**Comment by iamcodemaker** [comment]
Thought of another edge case we should handle. If two names only differ by the special characters they have in them, they could resolve to the same file names and overwrite eachother.

**Comment by iamcodemaker** [comment]
Updated to handle that edge case. I need to add some tests around this though.

**Comment by iamcodemaker** [comment]
The `createProfile()` function is unused. Should I remove it as well?

**Comment by iamcodemaker** [comment]
A thought I had, not for this PR, but maybe we should include default profiles for popular streaming services with Mixxx.

**Comment by daschuer** [comment]
> A thought I had, not for this PR, but maybe we should include default profiles for popular streaming services with Mixxx.

This is a free advertising for them. So we should not do this unless there is money involved.  We can ask them to publish demo profiles on their domains though.
Do you have one in mind? Maybe it is worth to ask them.

**Comment by iamcodemaker** [comment]
> This is a free advertising for them. So we should not do this unless there is money involved. We can ask them to publish demo profiles on their domains though.

Good point on the money. I had this thought as well. It would be better to charge money to be included as a default.

> Do you have one in mind? Maybe it is worth to ask them.

Yes, this came up in relation to https://cue.dj. Cue is a new (and beta) platform for streaming DJ sets. They post instructions for configuring Mixxx in their app and on their site.

**Comment by daschuer** [comment]
Thank you for taking care here. 

This solution creates the situation that we store redundant info in file name and profile name. I am unsure but I think that it is possible now to have duplicate profiles names.
Can the filename change during the lifetime of the profile object? 

Can we find an unambiguous way to tun the profile name into a file name? Maybe a Git like hash code or something? Or revert storing the profile name and use a save encoding like the url encoding?
QUrl::toPercentEncoding()

What do you think?
